### PR TITLE
Remove duplicate logic in aggregation node creation

### PR DESCRIPTION
**Pull request by tlento**

### CODE REVIEW

**Comment by tlento** [comment]
Current dependencies on/for this PR:
* main
  * **PR #235** <a href="https://app.graphite.dev/github/pr/transform-data/metricflow/235" target="_blank">Graphite</a> 
    * **PR #153** <a href="https://app.graphite.dev/github/pr/transform-data/metricflow/153" target="_blank">Graphite</a> 
      * **PR #236** <a href="https://app.graphite.dev/github/pr/transform-data/metricflow/236" target="_blank">Graphite</a> 
      * **PR #239** <a href="https://app.graphite.dev/github/pr/transform-data/metricflow/239" target="_blank">Graphite</a>  👈

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/transform-data/metricflow/239?utm_source=stack-comment).

File: metricflow/dataflow/builder/dataflow_plan_builder.py [hunk 1]
```diff
@@ -523,38 +523,16 @@ def build_aggregated_measures(
...
-            additive_measure_specs = grouped_measures_by_additiveness.additive_measures
-            grouped_semi_additive_measures = grouped_measures_by_additiveness.grouped_semi_additive_measures
+            measures_by_additiveness = grouped_measures_by_additiveness.measures_by_additiveness
 
-            # TODO: deduplicate this repeated logic
-            # Build output node for additive measures
-            if additive_measure_specs:
+            # Build output nodes for each distinct non-additive dimension spec
+            for measure_specs in measures_by_additiveness:
+                assert len(measure_specs) > 0, "received empy set of measure specs, this should not happen!"
```
Comment (WilliamDee) [inline metricflow/dataflow/builder/dataflow_plan_builder.py:530]: typo: `empy`

**Review by WilliamDee — APPROVED** [review]

File: metricflow/dataflow/builder/dataflow_plan_builder.py [hunk 2]
```diff
@@ -523,38 +523,16 @@ def build_aggregated_measures(
...
+                non_additive_spec = measure_specs[0].non_additive_dimension_spec
```
Comment (plypaul) [inline metricflow/dataflow/builder/dataflow_plan_builder.py:531]: Can you add a comment saying that `[0]` is fine since it's grouped?

**Comment by tlento** [comment]
I'm going to merge as-is and open a separate PR for the documentation stuff.

**Pull request merged**